### PR TITLE
Converts content cache tests to use pytest fixtures

### PR DESCRIPTION
**Pull request by dkliban**
[noissue]

### CODE REVIEW

File: pulp_file/tests/functional/api/from_pulpcore/test_content_cache.py [hunk 1]
```diff
@@ -18,162 +17,128 @@
...
+        r = get_url(url)
         if r.history:
             r = r.history[0]
-            return 200 if r.status_code == 302 else r.status_code, r.headers.get("X-PULP-CACHE")
-        return r.status_code, r.headers.get("X-PULP-CACHE")
+            return 200 if r.status == 302 else r.status, r.headers.get("X-PULP-CACHE")
+        return r.status, r.headers.get("X-PULP-CACHE")
+
+    # Sync from the remote and assert that a new repository version is created
+    remote = file_fixture_gen_remote(manifest_path=basic_manifest_path, policy="immediate")
```
Comment (gerrod3) [inline pulp_file/tests/functional/api/from_pulpcore/test_content_cache.py:49]: Same with this remote

File: pulp_file/tests/functional/api/from_pulpcore/test_content_cache.py [hunk 2]
```diff
@@ -18,162 +17,128 @@
...
+    ).created_resources
+    repo = file_repo_api_client.read(file_repo_with_auto_publish.pulp_href)
+    assert repo.latest_version_href.endswith("/versions/1/")
+
+    body = FileFilePublication(repository=repo.pulp_href)
+    pub2 = file_pub_api_client.read(
+        monitor_task(file_pub_api_client.create(body).task).created_resources[0]
+    )
+    response = file_distro_api_client.create(gen_distribution(repository=repo.pulp_href))
+    distro = file_distro_api_client.read(monitor_task(response.task).created_resources[0])
```
[the remark this answers is not in the document]
Comment (dkliban) [reply pulp_file/tests/functional/api/from_pulpcore/test_content_cache.py:62]: Yes! Thank you!

File: pulp_file/tests/functional/api/from_pulpcore/test_content_cache.py (hunk 1 above)
Comment (dkliban) [reply pulp_file/tests/functional/api/from_pulpcore/test_content_cache.py:49]: file_fixture_gen_remote already uses gen_object_with_cleanup

**Review by gerrod3 — APPROVED** [review]
Looks good.

**Pull request merged**